src/Form/NodeActionForm.php

```diff
@@ -0,0 +1,236 @@
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The tempstore.
+   *
+   * @var \Drupal\Core\TempStore\SharedTempStore
+   */
+  protected $tempStore;
+
+  /**
+   * The messenger service.
+   *
+   * @var \Drupal\Core\Messenger\MessengerInterface
+   */
+  protected $messenger;
+
+  /**
+   * The selection, in the entity_id => langcodes format.
+   *
+   * @var array
+   */
+  protected $selection = [];
+
+  /**
+   * The entity type definition.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeInterface
+   */
+  protected $entityType;
+
+  /**
+   * The action name.
+   *
+   * @var string
+   */
+  protected $action;
+
+  /**
+   * Constructs a new DeleteMultiple object.
+   *
+   * @param \Drupal\Core\Session\AccountInterface $current_user
+   *   The current user.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\TempStore\PrivateTempStoreFactory $temp_store_factory
+   *   The tempstore factory.
+   * @param \Drupal\Core\Messenger\MessengerInterface $messenger
+   *   The messenger service.
+   */
+  public function __construct(AccountInterface $current_user, EntityTypeManagerInterface $entity_type_manager, PrivateTempStoreFactory $temp_store_factory, MessengerInterface $messenger) {
+    $this->currentUser = $current_user;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->tempStore = $temp_store_factory->get('tide_node_action_multiple_confirm');
+    $this->messenger = $messenger;
+    $this->entityType = $this->entityTypeManager->getDefinition('node');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('current_user'),
+      $container->get('entity_type.manager'),
+      $container->get('tempstore.private'),
+      $container->get('messenger')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getQuestion() {
+    return $this->formatPlural(count($this->selection), 'Are you sure you want to ' . $this->action . ' this @item?', 'Are you sure you want to ' . $this->action . ' these @items?', [
+      '@item' => $this->entityType->getSingularLabel(),
+      '@items' => $this->entityType->getPluralLabel(),
+    ]);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCancelUrl() {
+    $route = Url::fromUserInput('/admin/content')->getRouteName();
+    return new Url($route);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'tide_node_action_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $ops = [];
+    $batch_size = 10;
+    foreach (array_chunk(array_keys($this->selection), $batch_size) as $ids) {
+      $ops[] = [get_class($this) . '::doAction', [$ids, $this->action]];
+    }
+    $batch = [
+      'title' => t('Processing all contents'),
```

Comment:
@vincent-gao I'd suggested changing this to
```suggestion
      'title' => t('Processing selected content'),
```
Content is both a singular and plural noun. Here's a Stack Overflow post with some more info. https://english.stackexchange.com/questions/45039/is-there-a-term-for-words-that-have-identical-singular-and-plural-forms

I used to work with the user that posted that question. What a small world.

Reply:
@GROwen Thanks for the review.I will take a look on `unpublish` requirement.